quad/src/qag_par.rs

```diff
@@ -372,13 +383,32 @@ impl QagPar {
                     new_result.1[k].3,
                 ));
             }
+            if {
```

Comment:
This `if` block is quite weird: define the condition before the `if`, or even in a separate function, avoiding the large block in this place (it's true that you can always replace a value with a block that returns that value, but in this position is definitely unexpected).